master/buildbot/reporters/gerrit.py

```diff
@@ -275,32 +275,28 @@ def stopService(self):
 
     @defer.inlineCallbacks
     def buildStarted(self, key, build):
-        if self.startCB is not None:
-            builder = yield self.master.data.get(("builders", build['builderid']))
-            build['builder'] = builder
-            if self.isBuildReported(build):
-                message = yield self.startCB(builder['name'], build, self.startArg)
-                self.sendCodeReviews(build, message)
-
-    @defer.inlineCallbacks
-    def buildFinished(self, builderName, build, result):
-        """Do the SSH gerrit verify command to the server."""
-        result = yield self.reviewCB(builderName, build, result, self.master, self.reviewArg)
-        result = _handleLegacyResult(result)
-        self.sendCodeReviews(build, result)
+        if self.startCB is None:
+            return
+        yield self.getBuildDetails(build)
```

Comment:
so we run "something" and that "something" has side effects?  if yes, I'd rather not :/